support-frontend/assets/components/onboarding/heading.tsx

```diff
@@ -102,71 +100,131 @@ const thankYouSubtext = css`
 	}
 
 	${from.desktop} {
-		margin-bottom: ${space[4]}px;
+		margin-bottom: ${space[8]}px;
 	}
 `;
 
-interface OnboardingHeadingProps extends CSSOverridable {
+interface OnboardingHeadingProps extends OnboardingProps {
 	onboardingStep: OnboardingSteps;
 }
 
-function OnboardingHeading({ onboardingStep }: OnboardingHeadingProps) {
-	// --------------------------- //
-	// TODO: Dynamically generated based on Products and User profile
-	const ONBOARDING_STEP_CONTENT_MAP: Record<
-		OnboardingSteps,
-		{ headingHero: string; heading: string; subtext: string }
-	> = {
-		[OnboardingSteps.Summary]: {
-			headingHero: 'aspect ratio 21:9',
-			heading: 'Thank you for subscribing to All-access digital',
-			subtext:
-				"You've just joined over 1.3m others who support independent journalism.",
-		},
-		[OnboardingSteps.GuardianApp]: {
-			headingHero: 'Placeholder',
-			heading: 'Placeholder',
-			subtext: 'Placeholder',
-		},
-		[OnboardingSteps.FeastApp]: {
-			headingHero: 'Placeholder',
-			heading: 'Placeholder',
-			subtext: 'Placeholder',
-		},
-		[OnboardingSteps.ThankYou]: {
-			headingHero: 'Placeholder',
-			heading: 'Placeholder',
-			subtext: 'Placeholder',
-		},
-	};
-	// --------------------------- //
+function OnboardingHeading({
+	onboardingStep,
+	landingPageSettings,
+	productKey,
+}: OnboardingHeadingProps) {
+	const order = getThankYouOrder();
+	const productSettings =
+		productKey && landingPageSettings.products[productKey];
+
+	const { windowWidthIsGreaterThan, windowWidthIsLessThan } = useWindowWidth();
+
+	const ONBOARDING_STEP_CONTENT_MAP = useMemo<
+		Record<
+			OnboardingSteps,
+			{
+				heading?: string;
+				subtext?: string;
+				gridId?: string;
+				altText?: string;
+				aspectRatio: SerializedStyles;
+			}
+		>
+	>(() => {
+		// Non breaking hyphen
+		const allAccessTitle =
+			productSettings?.title === 'All-access digital'
+				? `All\u2011access digital`
+				: productSettings?.title;
+
+		return {
+			[OnboardingSteps.Summary]: {
+				heading: `Thank you ${
+					order?.firstName && order.firstName + ' '
+				}for subscribing to ${allAccessTitle}`,
+				subtext:
+					"You've just joined over 1.3m others who support independent journalism.",
+				gridId: 'onboardingSummaryHero',
+				altText: 'Onboarding summary hero holding The Guardian logo',
```

Comment:
accessibility (post merge): 
this is a decorative image so the alt text should be empty
https://www.w3.org/WAI/tutorials/images/decorative/#:~:text=In%20these%20cases%2C%20a%20null%20(empty)%20alt%20text%20should%20be%20provided%20(alt%3D%22%22)%20so%20that%20they%20can%20be%20ignored%20by%20assistive%20technologies%2C%20such%20as%20screen%20readers.
```suggestion
				altText: '',
```

This also applies to the below ones.